src/ublox.cpp

```diff
@@ -206,50 +255,143 @@ void UBLOX::read_cb(uint8_t byte)
     if (decode_message())
     {
       parse_state_ = START;
+      if (message_class_ == CLASS_NAV && message_type_ == NAV_PVT)
+        last_pvt_timestamp = time_recieved;
+      if (new_data())
+        last_valid_message_ = millis();
     }
     else
     {
       // indicate error if it didn't work
       num_errors_++;
+      DBG("failed to parse message\n");
       parse_state_ = START;
     }
   }
 
   prev_byte_ = byte;
 }
 
-void UBLOX::get_pos_ecef(double* pos_ecef, uint32_t* t_ms)
+// convert a time struct to unix time
+// dumb leap years
+// This function will break in 2100, because it doesn't take into account that 2100 is not a leap year
+uint64_t convert_to_unix_time(const UBLOX::GNSS_TIME_T &time)
 {
-  (void)t_ms;
-  pos_ecef[0] = pos_ecef_message_.ecefX;
-  pos_ecef[1] = pos_ecef_message_.ecefY;
-  pos_ecef[2] = pos_ecef_message_.ecefZ;
+  uint32_t day_s = 24 * 60 * 60; // seconds per day
```

Comment:
can we use `gmtime` for this?
http://www.cplusplus.com/reference/ctime/gmtime/

Reply:
What was I thinking? One potential issue is that it usually, but not always uses unix time, at least before C++20. I don't know what else they would use.

Reply:
Change is made. It is much better.

Reply:
I don't see the update here.  Did you forget to push?